src/test/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobSchedulerTest.java

```diff
@@ -0,0 +1,318 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.Callback;
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import com.lithium.dbi.rdbi.testutil.TubeUtils;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import redis.clients.jedis.JedisPool;
+
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+@Test(groups = "integration")
+public class DedupJobSchedulerTest {
+    private static final RDBI rdbi = new RDBI(new JedisPool("localhost"));
+    private String tubeName;
+
+    private DedupJobScheduler scheduledJobSystem = null;
+
+    @BeforeMethod
+    public void setup(){
+        tubeName = TubeUtils.uniqueTubeName();
+        scheduledJobSystem  = new DedupJobScheduler(rdbi, "myprefix:");
+    }
+
+    @AfterMethod
+    public void tearDown(){
+        // nuke the queues
+        rdbi.withHandle(new Callback<Void>() {
+            @Override
+            public Void run(Handle handle) {
+                handle.jedis().del(scheduledJobSystem.getReadyQueue(tubeName), scheduledJobSystem.getRunningQueue(tubeName));
+                return null;
+            }
+        });
+
+        // ensure system is not paused
+        scheduledJobSystem.resume(tubeName);
+    }
+
+    @Test
+    public void testBasicSchedule() {
+        // Schedule a job.
+        scheduledJobSystem.schedule(tubeName, "{hello:world}", 0);
+
+        // Reserve the job. Should succeed
+        JobInfo result1 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertEquals(result1.getJobStr(), "{hello:world}");
+
+        // Try and reserve another job. Since nothing else to reserve should get back a null result.
+        JobInfo result2 = scheduledJobSystem.reserveSingle(tubeName, 1000);
+        assertNull(result2);
+    }
+
+    @Test
+    public void testBasicScheduleWhenPaused() {
+        scheduledJobSystem.pause(tubeName);
+
+        // Schedule a job. Should not be able to since the tube is paused.
+        assertFalse(scheduledJobSystem.schedule(tubeName, "{hello:world}", 0));
+    }
+
+    @Test
+    public void testScheduleSameJobMultipleTimesWithoutReserving() {
+        // Schedule a job.
+        assertTrue(scheduledJobSystem.schedule(tubeName, "{hello:world}", 0));
+
+        // Schedule same job again and verify it fails since the job is already in ready queue.
+        assertFalse(scheduledJobSystem.schedule(tubeName, "{hello:world}", 0));
```

Comment:
the return behavior has always been this way for our other schedulers. so i don't want to change the behavior here and be inconsistent. ill java doc more clearly.